### ISSUE TITLE
Crash Samsung s8

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#10983]
The following crash is produced when you resize a MapView (using TextureView) to full height and width:

```
 Process: com.mapbox.mapboxsdk.testapp, PID: 23309
 java.lang.Error: glDrawElements( static_cast<GLenum>(primitiveType), static_cast<GLsizei>(indexLength), GL_UNSIGNED_SHORT, reinterpret_cast<GLvoid*>(sizeof(uint16_t) * indexOffset)): 
   Error GL_OUT_OF_MEMORY at /Users/Nurbot/mapbox-gl-native/src/mbgl/gl/context.cpp:705
     at com.mapbox.mapboxsdk.maps.renderer.MapRenderer.nativeRender(Native Method)
     at com.mapbox.mapboxsdk.maps.renderer.MapRenderer.onDrawFrame(MapRenderer.java:90)
     at com.mapbox.mapboxsdk.maps.renderer.textureview.TextureViewMapRenderer.onDrawFrame(TextureViewMapRenderer.java:56)
 at com.mapbox.mapboxsdk.maps.renderer.textureview.TextureViewRenderThread.run(TextureViewRenderThread.java:282)
```

Until now I have only been able to reproduce this on a Samsung s8:

![ezgif com-video-to-gif 34](https://user-images.githubusercontent.com/2151639/35225988-2ed6497a-ff8a-11e7-9891-b1f6cb23c98d.gif)

Not reproducible on a pixel:
![ezgif com-video-to-gif 31](https://user-images.githubusercontent.com/2151639/35226010-4914f7a0-ff8a-11e7-87ae-676ac8aa6842.gif)

**Comment by tobrun** [comment]
Was able to detect that the line responsible for the crash is: `eglHolder#createSurface()` as part of:

```
        // If the surface size has changed inform the map renderer.
        if (recreateSurface) {
          eglHolder.createSurface();
          mapRenderer.onSurfaceChanged(gl, w, h);
          continue;
        }
```

When committing out that line, the map doesn't crash and on some devices work as expected but on another as Samsung J3 this results in not showing the surface when onStart/onStop is called in the hosting activity.

**Comment by tobrun** [comment]
fixed with #11148
